test/integration/targets/aws_secret/tasks/main.yaml

```diff
@@ -0,0 +1,285 @@
+      name: "hello-world-{{ resource_prefix }}"
+      state: present
+      zip_file: "{{ tmp.path }}/ansible_testing/hello_world.zip"
+      runtime: 'python2.7'
+      role: "arn:aws:iam::{{ test_caller_facts.account }}:role/test-secrets-manager-role-{{ resource_prefix }}"
+      handler: 'hello_world.lambda_handler'
+
+  - name: lambda policy for secrets manager
+    lambda_policy:
+      <<: *aws_connection_info
+      state: present
+      function_name: "hello-world-{{ resource_prefix }}"
+      statement_id: LambdaSecretsManagerTestPolicy
+      action: 'lambda:InvokeFunction'
+      principal: secretsmanager.amazonaws.com
+
+  # ============================================================
+  # Module parameter testing
+  # ============================================================
+  - name: test with no parameters
+    aws_secret:
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with no parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+      state: present
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  - name: test with missing parameters
+    aws_secret:
+      name: 'test_secret_string'
+      state: present
+      secret_type: 'string'
+    register: result
+    ignore_errors: true
+    check_mode: true
+
+  - name: assert failure when called with missing parameters
+    assert:
+      that:
+         - result.failed
+         - 'result.msg.startswith("missing required arguments:")'
+
+  # ============================================================
+  # Creation/Deletion testing
+  # ============================================================
+  - name: add secret to AWS Secrets Manager
+    aws_secret:
+      <<: *aws_connection_info
+      name: "test-secret-string-{{ resource_prefix }}"
+      state: present
+      secret_type: 'string'
+      secret: "{{ super_secret_string }}"
+    register: result
+
+  - name: assert correct keys are returned
+    assert:
+      that:
+        - result.changed
+        - result.ARN is not none
+        - result.Name is not none
+        - result.Tags is not none
+        - result.VersionIdsToStages is not none
+
+  - name: pausing for secret creation
+    pause:
+      seconds: 30
+
+  - name: no changes to secret
```

Comment:
Noob question:
What is the ansible-test command that allow me to run the integration test just for aws_secret ?

I'm a bit cheating locally and I guess it will be a source of differences in the result...

Reply:
`ansible-test integration aws_secret -v`

Reply:
@rrey This one is still making changes to the secret so the tests fail. Looks like an idempotence issue with the module. My PR restructuring things for CI has been merged and once your tests are completing rather than failing here the PR adding the permissions and this can be merged. Here's the output from testing on the dev server:
```
TASK [aws_secret : add secret to AWS Secrets Manager] **************************
changed: [testhost] => {"changed": true, "secret": {"arn": "arn:aws:secretsmanager:us-east-1:966509639900:secret:test-secret-string-ansible-test-macbook-pro-44-50736924-4CXvRp", "last_changed_date": "2019-02-21T19:44:25.772000+00:00", "name": "test-secret-string-ansible-test-shertel-50736924", "version_ids_to_stages": {"26d43df3-84a1-43eb-803e-40f0f0c65d1d": ["AWSCURRENT"]}}}

TASK [aws_secret : assert correct keys are returned] ***************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [aws_secret : no changes to secret] ***************************************
changed: [testhost] => {"changed": true, "secret": {"arn": "arn:aws:secretsmanager:us-east-1:966509639900:secret:test-secret-string-ansible-test-shertel-50736924-4CXvRp", "last_accessed_date": "2019-02-21T00:00:00+00:00", "last_changed_date": "2019-02-21T19:44:26.995000+00:00", "name": "test-secret-string-ansible-test-shertel-50736924", "version_ids_to_stages": {"26d43df3-84a1-43eb-803e-40f0f0c65d1d": ["AWSPREVIOUS"], "6ad8339b-27a9-4606-938e-8d2f0c5979eb": ["AWSCURRENT"]}}}

TASK [aws_secret : assert correct keys are returned] ***************************
fatal: [testhost]: FAILED! => {
    "assertion": "not result.changed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```